### PR TITLE
Throw error when user not found in ExperimentController

### PR DESCRIPTION
**Pull request by KD1712**

### CODE REVIEW

**Comment by RidhamShah** [comment]
This changes should be on all version of ExperimentClientController.

File: backend/packages/Upgrade/src/api/middlewares/UserCheckMiddleware.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+export class UserCheckMiddleware {
+  constructor(public settingService: SettingService, public experimentUserService: ExperimentUserService) {}
+
+  public async use(req: AppRequest, res: AppRequest, next: express.NextFunction): Promise<any> {
+    try {
+      const user_id = req.get('User-Id');
+      req.logger.child({ user_id });
+      req.logger.debug({ message: 'User Id is:', user_id });
+      console.log(user_id);
+      if (req.url !== '/api/init') {
```
Comment (RidhamShah) [inline backend/packages/Upgrade/src/api/middlewares/UserCheckMiddleware.ts:18]: There is no need to check userId from the header for the init call, so bring this if statement on the top.

**Review by RidhamShah — CHANGES_REQUESTED** [review]

**Comment by VivekFitkariwala** [comment]
@bcb37 @danoswaltCL, At present we are fetching the user doc in each HTTP request of Experiment Client and that is redundant. In this PR is moving that in the middleware and the error will be thrown at the place where the document is fetched instead of the service which is incorrect.
We can have this middleware in the latest v5 version and not touch the service for backward compatibility.

**Comment by danoswaltCL** [comment]
Another consideration is that this will break current non-client-library usage of the API, such as Livehint AI. They will need to begin sending this in the header for all calls.

**Comment by danoswaltCL** [comment]
Let's wait on merging this in, see my comments in https://github.com/CarnegieLearningWeb/UpGrade/pull/1606

**Review by VivekFitkariwala — APPROVED** [review]

**Comment by danoswaltCL** [comment]
holding off on this one, going with existing 404 errors in the individual endpoints still for now

**Comment by danoswaltCL** [comment]
Needs to be only a part of the v6 controller. see: https://github.com/CarnegieLearningWeb/UpGrade/issues/1851 @ppratikcr7 @VivekFitkariwala

**Comment by ppratikcr7** [comment]
Closing this as its handled in a new [PR](https://github.com/CarnegieLearningWeb/UpGrade/pull/1955) with v6 client controller.

**Pull request closed without merging**